combine_tool_outputs.R

```diff
@@ -58,10 +58,19 @@ suppressPackageStartupMessages(require(yaml))
 
 # parse input tables
 file_names = list.files(opt$input_dir, full.names=TRUE)
-predicted_labs_tables = lapply(file_names, function(file) read.csv(file, sep="\t", stringsAsFactors=FALSE))
+predicted_labs_tables = lapply(file_names, function(file) read.csv(file, sep="\t",
+                                                                   stringsAsFactors=FALSE,
+                                                                   comment.char = "#"))
 # extract datasets of origin
-file_names = sapply(file_names, function(p) file_path_sans_ext(basename(p)))
-datasets = sapply(file_names, function(name) unlist(strsplit(name, "_"))[1])
+tools = sapply(file_names, function(f) extract_metadata(f)[['tool']])
```

Comment:
Wouldn't it be more efficient to extract the complete dictionary of values from the header of each file, and then access that dictionary for e.g. tool and dataset? 

I know you're only doing it twice here, but I'm sure this pattern will expand in due course, and we might as well do it neatly from the start.

Reply:
@pinin4fjords definitely - thanks for spotting